src/Tagged.elm

```diff
@@ -1,94 +1,90 @@
 module Tagged exposing (..)
 
-{-|
+{-| A module that allows you to "tag" a value.
 
-A module that allows you to "tag" a value.
+@docs Tagged, tag, retag, untag, map, andMap, map2, andThen, extend
 
-@docs Tagged, tag, retag, untag, map, ap, map2, bind, extend
 -}
 
-{-|
-A Type that pairs a `value` with a `tag`.
+
+{-| A Type that pairs a `value` with a `tag`.
 
 The `tag` is ignored at runtime as evidenced by the only value constructor:
 
-```elm
-Tagged : value -> Tagged tag value
-```
+    Tagged : value -> Tagged tag value
+
 -}
 type Tagged tag value
-  = Tagged value
+    = Tagged value
 
-{-|
-An alias for the `Tagged` value constructor.
+
+{-| An alias for the `Tagged` value constructor.
 -}
 tag : value -> Tagged tag value
 tag =
-  Tagged
+    Tagged
+
 
-{-|
-Useful for composing functions together infix:
+{-| Useful for composing functions together in a pipeline:
+
+    foo =
+        Array.set |> map index |> andMap value |> andMap arr
 
-```elm
-foo =
-  Array.set `map` index `ap` value `ap` arr
-```
 -}
 map : (oldValue -> newValue) -> Tagged tag oldValue -> Tagged tag newValue
 map f (Tagged x) =
-  Tagged (f x)
+    Tagged (f x)
+
+
+{-| Useful for composing functions together in a pipeline:
 
-{-|
-Useful for composing functions together infix:
+    foo =
+        Array.set |> map index |> andMap value |> andMap arr
 
-```elm
-foo =
-  Array.set `map` index `ap` value `ap` arr
-```
 -}
-ap : Tagged tag (oldValue -> newValue) -> Tagged tag oldValue -> Tagged tag newValue
-ap (Tagged f) (Tagged x) =
-  Tagged (f x)
+andMap : Tagged tag (oldValue -> newValue) -> Tagged tag oldValue -> Tagged tag newValue
```

Comment:
Do we need to flip the args here?

Reply:
Right. This is turning out to be a rather lousy PR 😅